src/reason_pprint_ast.ml

```diff
@@ -2996,94 +3092,142 @@ class printer  ()= object(self:'self)
+        let leftWithOp = makeList ~postSpace:true [simplyFormatedLeftItm; atom updateToken] in
+        let expr = label ~space:true leftWithOp rightItm in
+        SpecificPrecedence ({reducePrecedence=tokenPrec; shiftPrecedence=tokenPrec}, expr)
+      )
+      | (Some simpleGet, None) -> (VeryHighPrecedence [simpleGet])
+      | (Some _, Some _) -> raise (NotPossible "Detected both simple sugar, and imperative set")
+      | (None, None) -> (
+        (* If there are any attributes, render unary like `(~-) x [@ppx]`, and infix like `(+) x y [@attr]` *)
+        let (arityAttrs, docAtrs, stdAttrs) = partitionAttributes x.pexp_attributes in
+        match (stdAttrs, printedStringAndFixityExpr e, ls) with
+        | ([], Infix printedIdent, [("", leftExpr); ("", rightExpr)]) ->
+          let infixToken = Token printedIdent in
+          let rightItm = self#ensureContainingRule ~withPrecedence:infixToken ~reducesAfterRight:rightExpr in
+          let leftItm = self#ensureExpression leftExpr ~reducesOnToken:infixToken in
+          let leftWithOp = makeList ~postSpace:true [leftItm; atom printedIdent] in
+          let indent = infixTokenRequiresIndent printedIdent in
+          let expr = label ~space:true ?indent leftWithOp rightItm in
+          SpecificPrecedence ({reducePrecedence=infixToken; shiftPrecedence=infixToken}, expr)
+        (* Will be rendered as `(+) a b c` which is parsed with higher precedence than all
+           the other forms unparsed here.*)
+        | ([], UnaryPlusPrefix printedIdent, [("", rightExpr)]) ->
+          let prec = Custom "prec_unary_plus" in
+          let rightItm = self#ensureContainingRule ~withPrecedence:prec ~reducesAfterRight:rightExpr in
+          let expr = label ~space:true (atom printedIdent) rightItm in
+          SpecificPrecedence ({reducePrecedence=prec; shiftPrecedence=Token printedIdent}, expr)
+        | ([], UnaryMinusPrefix printedIdent, [("", rightExpr)]) ->
+          let prec = Custom "prec_unary_minus" in
+          let rightItm = self#ensureContainingRule ~withPrecedence:prec ~reducesAfterRight:rightExpr in
+          let expr = label ~space:true (atom printedIdent) rightItm in
+          SpecificPrecedence ({reducePrecedence=prec; shiftPrecedence=Token printedIdent}, expr)
+        (* Will need to be rendered in self#expression as (~-) x y z. *)
+        | (_, _, _) ->
+          let attributesAsList = (List.map self#attribute stdAttrs) in
+          let theFunc = SourceMap (e.pexp_loc, (self#simple_expression e)) in
+          (*reset here only because [function,match,try,sequence] are lower priority*)
+          let theArgs = List.map self#reset#label_x_expression_param ls in
+          let theArgsAndAttrs = List.concat [theArgs; attributesAsList] in
+          VeryHighPrecedence (theFunc::theArgsAndAttrs)
+      )
+    )
+    (* TODO: Should protect this identifier *)
+    | Pexp_setinstvar (s, rightExpr) ->
```

Comment:
And then the rest of this function is the unparsing of other syntax elements such as setting of fields/record updates, or ternary.